lib/stack_master/sparkle_formation/compile_time/max_size_validator.rb

```diff
@@ -0,0 +1,36 @@
+require_relative 'value_validator'
+
+module StackMaster
+  module SparkleFormation
+    module CompileTime
+      class MaxSizeValidator < ValueValidator
+
+        KEY = :max_size
+
+        def initialize(name, definition, parameter)
+          @name = name
+          @definition = definition
+          @parameter = parameter
+        end
+
+        private
+
+        def check_is_valid
+          return true unless @definition[:type] == :number
+          return true unless @definition.key?(KEY)
+          invalid_values.empty?
+        end
+
+        def invalid_values
+          values = build_values(@definition, @parameter)
+          values.select {|value| value.to_f > @definition[KEY].to_f}
```

Comment:
Nitpick, it's idiomatic in Ruby to put spaces between curly braces `{ |value| ... }`